### PR TITLE
Connect Screen Designs

### LINKED ISSUE #1489: Update header of connect screen
The connect screen currently looks like:

Screenshot from 2020-04-13 23-15-54

This task is to update this screen to use the header component created in #1484 

Relevant design is here: https://www.figma.com/file/dzapXt5s0pKBxb9lZkIoDM/%231352---Confirm-transactions?node-id=916%3A261

The main file you will have to work in is `metamask-mobile/app/components/UI/AccountApproval/index.js`

### LINKED ISSUE #1498: Redesign "Connect" modal
The connect modal currently looks like:

Screenshot from 2020-04-20 14-58-36

This task is to redesign it so that it matches the new designs:

Screenshot from 2020-04-20 15-00-11

Designs: https://www.figma.com/file/dzapXt5s0pKBxb9lZkIoDM/%231352---Confirm-transactions?node-id=916%3A261

Relevant code is in `metamask-mobile/app/components/UI/AccountApproval/index.js`

### PR DESCRIPTION
**Pull request by EtDu**
**Description**

Implement connect screen (modal) redesigns. 

**Checklist**

- [x] `AccountApproval`: Use `TransactionHeader`
- [x] `AccountApproval`: Use `AccountInfoCard` 
- [x] `AccountApproval`: Update button style
- [x] `AccountApproval`: Change text layout

**Issue**

Resolves #1498 
Resolves #1489

### CODE REVIEW

**Comment by EtDu** [comment]
Screen Shot 2020-04-23 at 1 49 13 PM
@cjeria

**Comment by rickycodes** [comment]
@EtDu just checked this out (because I wanted to pull some work from it into my branch) and noticed this on my Pixel 3 running Android 10):

looks like we need a bit more space at the bottom:



additionally, I don't think we want to be able to scroll the account area:



I might tweak things a bit and push an update here shortly.

Otherwise, this is looking good to me :)

**Review by rickycodes — COMMENTED** [review]
looks good, just a couple spacing things I noticed on Android

**Comment by cjeria** [comment]
I agree with @rickycodes, let's remove scrolling, and can the tx container grow with content?

I also noticed a few font and font-color inconsistencies. See notes in the screenshot.  
image

**Comment by EtDu** [comment]
@cjeria the above is fixed. The font styles are correct on iOS, but for some reason they display this way on Android. I'm not sure how to fix this.

**Comment by rickycodes** [comment]
@EtDu don't worry about the font on android, I think it has something to do with it being the debug build? (I'm really not sure). IIRC the font is correct in the prod version, @cjeria and I have gotten tripped up by this before.

**Comment by estebanmino** [comment]
@cjeria are you sure is the old font? (I can't really differentiate those 😅). I don't think because is on debug mode the font will change, so maybe we have a bug there

File: app/components/UI/AccountApproval/index.js [hunk 1]
```diff
@@ -1,149 +1,59 @@
...
-		zIndex: 2
+	button: {
+		flex: 1
 	},
-	checkIcon: {
-		color: colors.white,
-		fontSize: 14,
-		lineHeight: 24
+	cancel: {
+		marginRight: 8
```
Comment (estebanmino) [inline app/components/UI/AccountApproval/index.js:53]: what about adding these margins to `actionContainer` that is the wrapper of this and `confirm`?

File: app/components/UI/AccountApproval/index.js [hunk 2]
```diff
@@ -1,149 +1,59 @@
...
 	},
-	icon: {
-		borderRadius: 27,
-		marginBottom: 12,
-		height: 54,
-		width: 54
+	confirm: {
+		marginLeft: 8
```
Comment (estebanmino) [inline app/components/UI/AccountApproval/index.js:56]: https://github.com/MetaMask/metamask-mobile/pull/1508/files#r423253606

**Review by estebanmino — COMMENTED** [review]
very small comment, LGTM

**Comment by cjeria** [comment]
Yes, I just added Robot font in Figma to verify and as you can see, it's indeed Roboto.

image

File: app/components/UI/AccountApproval/index.js (hunk 1 above)
Comment (EtDu) [reply app/components/UI/AccountApproval/index.js:53]: This is for the space in between the two buttons, not between the buttons and the screen edges.

**Review by estebanmino — APPROVED** [review]
LGTM!

**Pull request merged**